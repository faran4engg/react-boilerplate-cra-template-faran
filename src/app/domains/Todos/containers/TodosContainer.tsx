import { FC, memo } from 'react';

import { getAllTodos } from './queries';

import { OwnProps, RenderProps } from './types';
import { useQuery } from 'react-query';

const TodosContainer: FC<OwnProps & RenderProps> = ({ children }) => {
  const { isLoading, error, data } = useQuery('todos', getAllTodos, {
    refetchOnWindowFocus: false,
    suspense: true,
  });

  if (isLoading) return <h1>Loading...'</h1>;

  if (error) return <p>An error has occurred'</p>;

  return (
    <div className="text-gray-600 dark:text-gray-400">
      {children({
        isLoading: false,
        todos: data,
      })}
    </div>
  );
};

export default memo(TodosContainer);
