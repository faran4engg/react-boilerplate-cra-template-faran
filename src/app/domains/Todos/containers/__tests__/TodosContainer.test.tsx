import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import TodosContainer from '..';

jest.mock('react-query', () => ({
  useQuery: () => ({}),
}));

describe('<TodosContainer />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render and match the snapshot when loading', () => {
    jest.mock('react-query', () => ({
      useQuery: () => ({
        isLoading: true,
        error: undefined,
        data: undefined,
      }),
    }));

    const renderer = createRenderer();

    renderer.render(<TodosContainer>{() => <div />}</TodosContainer>);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render and match the snapshot when error', () => {
    jest.mock('react-query', () => ({
      useQuery: () => ({
        isLoading: false,
        error: {},
        data: undefined,
      }),
    }));
    const renderer = createRenderer();

    renderer.render(<TodosContainer>{() => <div />}</TodosContainer>);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
  it('should render and match the snapshot successfully', () => {
    jest.mock('react-query', () => ({
      useQuery: () => ({
        isLoading: false,
        error: undefined,
        data: [{}],
      }),
    }));
    const renderer = createRenderer();
    renderer.render(<TodosContainer>{() => <div />}</TodosContainer>);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
