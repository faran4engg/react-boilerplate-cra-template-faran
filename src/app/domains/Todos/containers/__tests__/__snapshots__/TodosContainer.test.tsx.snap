// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodosContainer /> should render and match the snapshot successfully 1`] = `
<div
  className="text-gray-600 dark:text-gray-400"
>
  <div />
</div>
`;

exports[`<TodosContainer /> should render and match the snapshot when error 1`] = `
<div
  className="text-gray-600 dark:text-gray-400"
>
  <div />
</div>
`;

exports[`<TodosContainer /> should render and match the snapshot when loading 1`] = `
<div
  className="text-gray-600 dark:text-gray-400"
>
  <div />
</div>
`;
